### PR TITLE
Update README.md for installation latest Kedro

### PR DESCRIPTION
**Pull request by noklam**
## Description

Add a section for installation from source. We used to have nightly build(?), this is mainly for advance user who are willing to try out the latest Kedro which is not released yet. It would be nice to have a section to point them to.

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,6 +37,14 @@ conda install -c conda-forge kedro
 
 Our [Get Started guide](https://docs.kedro.org/en/stable/get_started/install.html) contains full installation instructions, and includes how to set up Python virtual environments.
 
+### Installation from source
+To install Kedro from source code, clone the repository and run:
```
Comment (stichbury) [inline README.md:41]:
```suggestion
To install Kedro from source code, clone the repository:
```

File: README.md [hunk 2]
```diff
@@ -37,6 +37,14 @@ conda install -c conda-forge kedro
...
+```
+git clone git@github.com:kedro-org/kedro.git
+cd kedro
+pip install .
+```
```
Comment (stichbury) [inline README.md:46]: Can I suggest that you split this into two, so the `git clone` command, then the text "Change to the `kedro` directory and enter the following: `pip install .`"

File: README.md (hunk 1 above)
Comment (stichbury) [inline README.md:41]: Maybe explain why you'd want to do this in a simple sentence?

**Review by stichbury — CHANGES_REQUESTED** [review]
Few minor suggestions but it looks good to me.

**Comment by noklam** [comment]
Thanks, add a sentence to explain why is it useful.

File: README.md [hunk 3]
```diff
@@ -37,6 +37,17 @@ conda install -c conda-forge kedro
...
+### Installation from source
+To access the latest Kedro version before its official release, install it from the source code. First, clone the repository:
+```
+git clone git@github.com:kedro-org/kedro.git
+```
+
+Then change to the `kedro` directory and enter the following:
+```
+pip install .
+```
```
Comment (astrojuanlu) [inline README.md:49]: What about

```
pip install git+https://github.com/kedro-org/kedro
```

?

Comment (astrojuanlu) [reply README.md:49]: (To avoid the intermediate cloning step)

Comment (noklam) [reply README.md]: sounds good

File: README.md [hunk 4]
```diff
@@ -37,6 +37,12 @@ conda install -c conda-forge kedro
 
 Our [Get Started guide](https://docs.kedro.org/en/stable/get_started/install.html) contains full installation instructions, and includes how to set up Python virtual environments.
 
+### Installation from source
+To access the latest Kedro version before its official release, install it from the `main` branch.
+```
```
Comment (stichbury) [inline README.md:42]:
```suggestion
```bash
```

**Review by stichbury — APPROVED** [review]
LGTM 🌟 Thanks @noklam

**Review by astrojuanlu — APPROVED** [review]

**Pull request merged**